docs/ble.md

```diff
@@ -58,36 +73,36 @@ dictionary CharacteristicInit {
     NotifyCallback onNotify;            // optional
 };
 
-callback ReadCallback = void (unsigned long offset, FulfillReadCallback);
+interface Characteristic {
+    attribute ReadCallback onReadRequest;
+    attribute WriteCallback onWriteRequest;
+    attribute SubscribeCallback onSubscribe;
+    attribute UnsubscribeCallback onUnsubscribe;
+    attribute NotifyCallback onNotify;
+	attribute CharacteristicResult response;
+};
+
+callback ReadCallback = void (unsigned long offset,
+                              FulfillReadCallback fulfillreadcallback);
 callback WriteCallback = void (Buffer data, unsigned long offset,
-                               boolean withoutResponse, FulfillWriteCallback);
+                               boolean withoutResponse,
+							   FulfillWriteCallback fulfillwritecallback);
 callback SubscribeCallback = void (unsigned long maxValueSize,
-                                   FulfillSubscribeCallback);
+                                   FulfillSubscribeCallback fullfillsubscribecallback);
 callback FulfillReadCallback = void (CharacteristicResult result, Buffer data);
 callback FulfillWriteCallback = void (CharacteristicResult result);
 callback FulfillSubscribeCallback = void (Buffer data);
 
+enum CharacteristicResult { "RESULT_SUCCESS", "RESULT_INVALID_OFFSET", "RESULT_INVALID_ATTRIBUTE_LENGTH", "RESULT_UNLIKELY_ERROR" } ;
+
 dictionary DescriptorInit {
     string uuid;
     string value;
 };
-
-[NoInterfaceObject]
-interface Characteristic {
-    attribute ReadCallback onReadRequest;
-    attribute WriteCallback onWriteRequest;
-    attribute SubscribeCallback onSubscribe;
-    attribute UnsubscribeCallback onUnsubscribe;
-    attribute NotifyCallback onNotify;
-    unsigned long RESULT_SUCCESS;
-    unsigned long RESULT_INVALID_OFFSET;
-    unsigned long RESULT_INVALID_ATTRIBUTE_LENGTH;
-    unsigned long RESULT_UNLIKELY_ERROR;
-};
 ```
 
-API Documentation
------------------
+BLE-supported Events
```

Comment:
The term "BLE-supported" doesn't make sense to me. See how they list events in the [Node dgram API](https://nodejs.org/dist/latest-v8.x/docs/api/dgram.html) - they go in the same "Class: xyz" section as the APIs below.

Reply:
@grgustaf -- my tool could easily(!) be adapted to check for WebIDL syntax.  In my copious spare time, I'll gen up a simple app -- but I'll need someone who knows (useful) things to put it into the automatic testing...

...If that's what we want; I had sort of just planned on periodically running my actual tool over the code as a natural result of my testing/development and keeping the WebIDL up to date that way...

Reply:
In re: events (i.e., "ble-supported" events)

The problem here is that each event isn't really part of the BLE object; in some ways, it's bifurcating the documentation: on one hand, most of the docs describe the interface the user sees/manipulates, etc. -- but with events (and maybe other things?), it's really addressing concerns of the guys implementing the underlying behavior.  I don't know that the text describing events even belongs in this documentation...(?)

Reply:
This might be a misunderstanding? Those events are indeed part of the BLE object and the ZJS user can (and needs to) register functions to listen for them with the .on() method. For example, see this [BLE sample](https://github.com/t-harvey/zephyr.js/blob/master/samples/WebBluetoothGroveLcdDemo.js#L161).

Reply:
The `on()` method comes in because BLE is an EventEmitter; perhaps this needs to be made more clear, but it does basically behave like Node now and this was an attempt to document things in the same way they do. Maybe we missed some connecting glue though. :)